### PR TITLE
Convert the codebase to TypeScript, use Prettier and update linting setup

### PR DESCRIPTION
**Pull request by charlesdemers**
This makes the addon on-par with what we do in our projects: 

- TypeScript + tslint
- Prettier
- Updated eslint config

### CODE REVIEW

File: addon/services/best-language.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+interface Language {
+  language: string;
+  baseLanguage: string;
+  score: number;
+}
+
+interface LanguageWithScore {
+  language: string;
```
Comment (gingman) [inline addon/services/best-language.ts]: Ça sert à quoi `LanguageWithScore` si on a déjà le score dans `Language`?

**Review by gingman — APPROVED** [review]

File: addon/services/best-language.ts (hunk 1 above)
Comment (charlesdemers) [reply addon/services/best-language.ts]: It’s the type we use before having `baseLanguage` in the object, I’m open to suggestions if you have a better name though!

Comment (gingman) [reply addon/services/best-language.ts]: We could only have a `Language` interface with the `baseLanguage` optional.

Comment (charlesdemers) [reply addon/services/best-language.ts]: You’re right 🤦‍♂️ I need to step up my TS game! 😄

**Pull request merged**